script.js

```diff
@@ -0,0 +1,336 @@
+document.addEventListener("DOMContentLoaded", function () {
+    //Elements
+    const quizQuestions = [];
+    const firstPage = document.getElementById('first-page');
+    const form = document.getElementById('quiz-form');
+    const select = form.querySelector('select');
+    const questionsList = document.getElementById('questions-list');
+    const quizDisplay = document.getElementById('quiz-display');
+    const dialog = document.getElementById('dialog');
+    const playerNamesForm = document.getElementById('player-names-form');
+    const player1 = document.getElementById('player1-name');
+    const player2 = document.getElementById('player2-name');
+    const quizListContainer = document.getElementById('quiz-list-container');
+    const startQuizButton = document.getElementById('startQuiz');
+    const player1ScoreDisplay = document.getElementById('player1-points');
+    const player2ScoreDisplay = document.getElementById('player2-points');
+    const player1ScoreUpButton = document.getElementById('player1-score-up');
+    const player1ScoreDownButton = document.getElementById('player1-score-down');
+    const player2ScoreUpButton = document.getElementById('player2-score-up');
+    const player2ScoreDownButton = document.getElementById('player2-score-down');
+    const currentPlayerDisplay = document.getElementById('current-player');
+
+    //Variables
+    let currentPlayer = 1;
+    let currentQuestionIndex = 0;
+    let player1Score = 0;
+    let player2Score = 0;
+
+    //Event listeners for score button
+    player1ScoreUpButton.addEventListener('click', function () {
+      updateScore('player1', 1);
+    });
+
+    player1ScoreDownButton.addEventListener('click', function () {
+      updateScore('player1', -1);
+    });
+
+    player2ScoreUpButton.addEventListener('click', function () {
+      updateScore('player2', 1);
+    });
+
+    player2ScoreDownButton.addEventListener('click', function () {
+      updateScore('player2', -1);
+    });
+
+    //Function to update players score
+    function updateScore(player, change) {
+      const scoreDisplay = player === 'player1' ? player1ScoreDisplay : player2ScoreDisplay;
+      if (player === 'player1') {
+        player1Score += change;
+      } else {
+        player2Score += change;
+      }
+      scoreDisplay.textContent = player === 'player1' ? player1Score : player2Score;
+    }
+
+    //Event listeners for start quiz button
+    startQuizButton.addEventListener('click', function () {
+        quizListContainer.style.display = 'none';
```

Comment:
You could refactor this to a separate function: maybe the lines responsible for hiding elements could go to one funcion and the rest, like player name assignments to the `startQuiz` fuction. Ideally you want your code to read like a natural language. So the `startQuiz` would be the event handler - funcion you call when the event happens, and `startQuiz` would call funcions like `setPlayersNames`, `displayQuizElements` or something in that fashion. Then from the code we can learn what steps are necessary to start the quiz.